### PR TITLE
make output format configurable

### PR DESCRIPTION
**Pull request by casimir**
A common pratice for linters is to provides a way to format output. This ease integration with editors and usablity in scripts. I don't know how you feel about adding formatting support to gometalinter so I made a basic implementation to discuss it.

This add a `--format` option. It uses a format string and `text/template` from the stdlib to render an issue line. The default format is retrocompatible with the previous behavior. I couldn't find a significant impact on performance.
What still missing is a basic documentation, at least the available fields.

What do you think?

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -225,6 +226,7 @@ var (
 	checkstyleFlag      = kingpin.Flag("checkstyle", "Generate checkstyle XML rather than standard line-based output.").Bool()
 	enableGCFlag        = kingpin.Flag("enable-gc", "Enable GC for linters (useful on large repositories).").Bool()
 	aggregateFlag       = kingpin.Flag("aggregate", "Aggregate issues reported by several linters.").Bool()
+	formatFlag          = kingpin.Flag("format", "Output format.").Default("{{.Path}}:{{.Line}}:{{if gt .Col 0}}{{.Col}}{{end}}:{{.Severity}}: {{.Message}} ({{.Linter}})").String()
```
Comment (alecthomas) [inline main.go]: ``{{if gt .Col 0}}` can just be `{{if .Col}}` due to template truthiness

File: main.go [hunk 2]
```diff
@@ -260,11 +262,12 @@ type Issue struct {
 }
 
 func (i *Issue) String() string {
-	col := ""
-	if i.Col != 0 {
-		col = fmt.Sprintf("%d", i.Col)
-	}
-	return fmt.Sprintf("%s:%d:%s:%s: %s (%s)", strings.TrimSpace(i.Path), i.Line, col, i.Severity, strings.TrimSpace(i.Message), i.Linter)
+	tmpl, err := template.New("output").Parse(*formatFlag)
```
Comment (alecthomas) [inline main.go]: It would be better to make this a global, possibly via an `Action()` on the formatFlag.

**Comment by alecthomas** [comment]
Seems like a reasonable idea; just a couple of changes.

**Comment by casimir** [comment]
I tried to adress your comments with the last commit. I'm not familiar with _kingpin_, feel free to tell me if I did something wrong.

**Comment by alecthomas** [comment]
Perfect, thanks!

**Comment by casimir** [comment]
Well you're fast! Glad it helps.

**Pull request merged**